crates/vqueues/src/delay_queue.rs

```diff
@@ -0,0 +1,1382 @@
+// Based on a copy of DelayQueue from tokio-util
+// https://github.com/tokio-rs/tokio/tree/4714ca168d6bd97193625657b0381e9b65a9ceff/tokio-util/src/time
```

Comment:
Did you apply any changes to the copy? If not, then I would skip the detailed review.